modules/blocks/verify.js

```diff
@@ -176,33 +187,59 @@ Verify.prototype.verifyBlock = function (block) {
 		try {
 			bytes = library.logic.transaction.getBytes(transaction);
 		} catch (e) {
-			result.errors.push(e.toString());
+			return e.toString();
 		}
 
 		if (appliedTransactions[transaction.id]) {
-			result.errors.push('Encountered duplicate transaction: ' + transaction.id);
+			return 'Encountered duplicate transaction: ' + transaction.id;
 		}
 
 		appliedTransactions[transaction.id] = transaction;
 		if (bytes) { payloadHash.update(bytes); }
 		totalAmount += transaction.amount;
 		totalFee += transaction.fee;
 	}
-
+	
 	if (payloadHash.digest().toString('hex') !== block.payloadHash) {
-		result.errors.push('Invalid payload hash');
+		return 'Invalid payload hash';
 	}
 
 	if (totalAmount !== block.totalAmount) {
-		result.errors.push('Invalid total amount');
+		return 'Invalid total amount';
 	}
 
 	if (totalFee !== block.totalFee) {
-		result.errors.push('Invalid total fee');
+		return 'Invalid total fee';
+	}
+	
+	// Calculate expected rewards
+	var expectedReward = __private.blockReward.calcReward(block.height);
+
+	if (block.height !== 1 && expectedReward !== block.reward && exceptions.blockRewards.indexOf(block.id) === -1) {
+		return ['Invalid block reward:', block.reward, 'expected:', expectedReward].join(' ');
+	}
+
+	var valid;
+	try {
+		valid = library.logic.block.verifySignature(block);
+	} catch (e) {
+		return e.toString();
+	}
+	if (!valid) {
+		return 'Failed to verify block signature';
+	}
+	
+	var getBlockId;
+	try {
+		getBlockId = library.logic.block.getId(block);
+	} catch (e) {
+		return e.toString();
 	}
 
-	result.verified = result.errors.length === 0;
-	return result;
+	if (getBlockId !== block.id) {
```

Comment:
Ok, in this way `verifyBlock` will validate parameters and at the end generate and add `block.id`